cmd/controller-manager/app/config.go

```diff
@@ -17,17 +17,177 @@ limitations under the License.
+	ClusterSigningKeyFile  string
+	ClusterSigningCertFile string
+}
+
+// DeploymentControllerInfo is part of context object for the controller manager.
+type DeploymentControllerInfo struct {
+	ConcurrentDeploymentSyncs      int32
+	DeploymentControllerSyncPeriod metav1.Duration
+}
+
+// PodGCControllerInfo is part of context object for the controller manager.
+type PodGCControllerInfo struct {
+	TerminatedPodGCThreshold int32
+}
+
+// EndPointControllerInfo is part of context object for the controller manager.
+type EndPointControllerInfo struct {
+	ConcurrentEndpointSyncs int32
+}
+
+// GarbageCollectorControllerInfo is part of context object for the controller manager.
+type GarbageCollectorControllerInfo struct {
+	ConcurrentGCSyncs      int32
+	GCIgnoredResources     []componentconfig.GroupResource
+	EnableGarbageCollector bool
+}
+
+// HPAControllerInfo is part of context object for the controller manager.
+type HPAControllerInfo struct {
+	HorizontalPodAutoscalerUseRESTClients           bool
+	HorizontalPodAutoscalerTolerance                float64
+	HorizontalPodAutoscalerDownscaleForbiddenWindow metav1.Duration
+	HorizontalPodAutoscalerUpscaleForbiddenWindow   metav1.Duration
+	HorizontalPodAutoscalerSyncPeriod               metav1.Duration
+}
+
+// NamespaceControllerInfo is part of context object for the controller manager.
+type NamespaceControllerInfo struct {
+	NamespaceSyncPeriod      metav1.Duration
+	ConcurrentNamespaceSyncs int32
+}
+
+// NodeLifecycleControllerInfo is part of context object for the controller manager.
+type NodeLifecycleControllerInfo struct {
+	EnableTaintManager        bool
+	NodeEvictionRate          float32
+	SecondaryNodeEvictionRate float32
+	NodeStartupGracePeriod    metav1.Duration
+	NodeMonitorGracePeriod    metav1.Duration
+	PodEvictionTimeout        metav1.Duration
+	LargeClusterSizeThreshold int32
+	UnhealthyZoneThreshold    float32
+}
+
+// PersistentVolumeBinderControllerInfo is part of context object for the controller manager.
+type PersistentVolumeBinderControllerInfo struct {
+	PVClaimBinderSyncPeriod metav1.Duration
+	VolumeConfiguration     componentconfig.VolumeConfiguration
+}
+
+// ReplicaSetInfo is part of context object for the controller manager.
+type ReplicaSetInfo struct {
+	ConcurrentRSSyncs int32
+}
+
+// ReplicationControllerInfo is part of context object for the controller manager.
+type ReplicationControllerInfo struct {
+	ConcurrentRCSyncs int32
+}
+
+// ResourceQuotaControllerInfo is part of context object for the controller manager.
+type ResourceQuotaControllerInfo struct {
+	ResourceQuotaSyncPeriod      metav1.Duration
+	ConcurrentResourceQuotaSyncs int32
+}
+
+// SAControllerInfo is part of context object for the controller manager.
+type SAControllerInfo struct {
+	ConcurrentSATokenSyncs int32
+}
+
+// ComponentConfigInfo is the main context object for the controller manager.
+type ComponentConfigInfo struct {
+	GenericComponentConfig                 GenericComponentInfo
```

Comment:
do we need the Config postfix in those fields?

Reply:
Now that we remove those struct into component api, delete postfix `Config` now